zerver/tests/test_decorators.py

```diff
@@ -2061,3 +2062,55 @@ def clear_cache() -> None:
         self.assertEqual(hits, 0)
         self.assertEqual(misses, 0)
         self.assertEqual(currsize, 0)
+
+
+class TestRequestNotes(ZulipTestCase):
+    def test_request_notes_realm(self) -> None:
+        """
+        This test verifies that .realm gets set correctly on the request notes
+        depending on the subdomain.
+        """
+
+        def mock_home(expected_realm: Optional[Realm]) -> Callable[[HttpRequest], HttpResponse]:
+            def inner(request: HttpRequest) -> HttpResponse:
+                self.assertEqual(RequestNotes.get_notes(request).realm, expected_realm)
+                return HttpResponse()
+
+            return inner
+
+        zulip_realm = get_realm("zulip")
+        with mock.patch("zerver.middleware.get_subdomain", return_value="zulip"), mock.patch(
```

Comment:
We can just pass `subdomain` to `client_get` instead of this mocking strategy.

It's implemented in this function:
```
    def set_http_headers(self, kwargs: Dict[str, Any]) -> None:                       
        if "subdomain" in kwargs:                                                     
            kwargs["HTTP_HOST"] = Realm.host_for_subdomain(kwargs["subdomain"])       
            del kwargs["subdomain"]                                                   
        elif "HTTP_HOST" not in kwargs:                                               
            kwargs["HTTP_HOST"] = Realm.host_for_subdomain(self.DEFAULT_SUBDOMAIN)    
```

in `test_classes.py`; maybe we need to do a small documentation PR to add docstrings for the `api_get` / `client_get` type functions listing their features?

Reply:
Perhaps just including `subdomain` in the function signature? Like `def client_get(self, url: str, info: Dict[str, Any] = {}, subdomain: str="zulip", **kwargs: Any) -> HttpResponse:`

I feel that the argument missing from the signature is what makes me forget about its existence

Reply:
Yeah, that'd be pretty reasonable, though might complicate the implementation, given the way we pass around `kwargs`.  Possibly the right answer is to just name the ~5 parameters that make up kwargs explicitly?  It's kinda annoying in that we have approximately 10 functions that accept this `kwargs` bundle.  I guess `subdomain` is the main one that's used broadly; so I think it'd be reasonable to move only it out of `kwargs` and not the rest.